From 2157720bbafa076800d4ee0793357815c45001f0 Mon Sep 17 00:00:00 2001
From: Ji Luo <ji.luo@nxp.com>
Date: Wed, 25 Jan 2023 15:58:11 +0800
Subject: [PATCH 1/3] MA-21007 integrate libtrustymatter

the libtrustymatter third-party lib contains the Trusty IPC
infrastructure and client application in the Non-secure side.

Signed-off-by: Ji Luo <ji.luo@nxp.com>
---
 .gitmodules                          |  4 ++-
 third_party/libtrustymatter/BUILD.gn | 47 ++++++++++++++++++++++++++++
 third_party/libtrustymatter/repo     |  1 +
 3 files changed, 51 insertions(+), 1 deletion(-)
 create mode 100644 third_party/libtrustymatter/BUILD.gn
 create mode 160000 third_party/libtrustymatter/repo

diff --git a/.gitmodules b/.gitmodules
index 7117074816..fab5f20fe5 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -284,4 +284,6 @@
         url = https://github.com/bouffalolab/bl_iot_sdk_tiny.git
         branch = master
         platforms = bouffalolab
-
+[submodule "third_party/libtrustymatter/repo"]
+	path = third_party/libtrustymatter/repo
+	url = http://androidsource.ap.freescale.net/project/a/libtrustymatter
diff --git a/third_party/libtrustymatter/BUILD.gn b/third_party/libtrustymatter/BUILD.gn
new file mode 100644
index 0000000000..966505e4e8
--- /dev/null
+++ b/third_party/libtrustymatter/BUILD.gn
@@ -0,0 +1,47 @@
+# Copyright (c) 2023 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build_overrides/chip.gni")
+
+config("libtrustymatter_config") {
+  include_dirs = [
+    "repo/matter_client/include",
+    "repo/libtrusty/include",
+  ]
+
+  cflags = [ "-Wno-implicit-fallthrough" ]
+  cflags_cc = [ "-std=c++17" ]
+}
+
+source_set("libtrustymatter") {
+  sources = [
+    "repo/matter_client/trusty_matter.cpp",
+    "repo/matter_client/trusty_matter_ipc.cpp",
+    "repo/matter_client/matter_messages.cpp",
+    "repo/matter_client/serializable.cpp",
+    "repo/matter_client/include/mem.h",
+    "repo/matter_client/include/matter_defs.h",
+    "repo/matter_client/include/matter_ipc.h",
+    "repo/matter_client/include/matter_messages.h",
+    "repo/matter_client/include/serializable.h",
+    "repo/matter_client/include/trusty_matter.h",
+    "repo/matter_client/include/trusty_matter_ipc.h",
+    "repo/matter_client/include/UniquePtr.h",
+    "repo/libtrusty/trusty.c",
+    "repo/libtrusty/include/trusty/ipc.h",
+    "repo/libtrusty/include/trusty/tipc.h",
+  ]
+
+  public_configs = [ ":libtrustymatter_config" ]
+}
diff --git a/third_party/libtrustymatter/repo b/third_party/libtrustymatter/repo
new file mode 160000
index 0000000000..df112aa09f
--- /dev/null
+++ b/third_party/libtrustymatter/repo
@@ -0,0 +1 @@
+Subproject commit df112aa09f50610c67020463c4dbb39fbdac4c8f
-- 
2.25.1

